src/distribute_kv_cache/rpc/client.rs

```diff
@@ -0,0 +1,779 @@
+            client_id,
+        }
+    }
+
+    /// Recv request header from the stream
+    pub async fn recv_header(&self) -> Result<RespHeader, RpcError> {
+        // Try to read to buffer
+        match self.recv_len(REQ_HEADER_SIZE).await {
+            Ok(()) => {}
+            Err(err) => {
+                debug!("Failed to receive request header: {:?}", err);
+                return Err(err);
+            }
+        }
+
+        let buffer: &mut BytesMut = unsafe { &mut *self.resp_buf.get() };
+        let req_header = RespHeader::decode(buffer)?;
+        debug!("Received response header: {:?}", req_header);
+
+        Ok(req_header)
+    }
+
+    /// Receive request body from the server.
+    pub async fn recv_len(&self, len: u64) -> Result<(), RpcError> {
+        let start = tokio::time::Instant::now();
+        let mut req_buffer: &mut BytesMut = unsafe { &mut *self.resp_buf.get() };
+        if req_buffer.capacity() < u64_to_usize(len) {
+            req_buffer.reserve(u64_to_usize(len) + 1);
+            debug!(
+                "Client reserve buffer {:?} to size {:?} cost: {:?}",
+                u64_to_usize(len),
+                len,
+                start.elapsed()
+            );
+        }
+        // req_buffer.resize(u64_to_usize(len), 0);
+        // req_buffer.resize(u64_to_usize(len), 0);
+        unsafe {
+            req_buffer.set_len(u64_to_usize(len));
+        }
+        let start_1 = start.elapsed();
+        debug!(
+            "Client resize buffer {:?} to size {:?} cost: {:?}",
+            u64_to_usize(len),
+            len,
+            start_1
+        );
+
+        let reader = self.get_stream_mut();
+        match read_exact_timeout!(reader, &mut req_buffer, self.timeout_options.read_timeout).await
+        {
+            Ok(size) => {
+                debug!("{:?} Received response body len: {:?}", self, size);
+                Ok(())
+            }
+            Err(err) => {
+                debug!("{:?} Failed to receive response header: {:?}", self, err);
+                Err(RpcError::InternalError(err.to_string()))
+            }
+        }
+    }
+
+    /// Receive request body from the server, try to read huge len data with given buffer.
+    pub async fn recv_huge_len(&self, len: u64, req_buffer: &mut BytesMut) -> Result<(), RpcError> {
+        let start = tokio::time::Instant::now();
+        if req_buffer.capacity() < u64_to_usize(len) {
+            req_buffer.reserve(u64_to_usize(len) + 1);
+            debug!(
+                "Client reserve buffer {:?} to size {:?} cost: {:?}",
+                u64_to_usize(len),
+                len,
+                start.elapsed()
+            );
+        }
+        // req_buffer.resize(u64_to_usize(len), 0);
+        // req_buffer.resize(u64_to_usize(len), 0);
+        unsafe {
+            req_buffer.set_len(u64_to_usize(len));
+        }
+        let start_1 = start.elapsed();
```

Comment:
this line is useless

Reply:
Got, it is used for benchmark and can be removed safely.